e2e/sidecars/sidecars_test.go

```diff
@@ -0,0 +1,273 @@
+			apps, err := cli.AppList()
+			Expect(err).NotTo(HaveOccurred())
+			Expect(apps).To(ContainSubstring(appName))
+		})
+
+		It("app show includes app name", func() {
+			appShowOutput, err := cli.AppShow(appName)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(appShowOutput.Name).To(Equal(appName))
+			Expect(appShowOutput.URI).To(BeEmpty())
+		})
+	})
+
+	Context("when creating a new environment", func() {
+		var (
+			testEnvInitErr error
+		)
+		BeforeAll(func() {
+			_, testEnvInitErr = cli.EnvInit(&client.EnvInitRequest{
+				AppName: appName,
+				EnvName: "test",
+				Profile: "default",
+				Prod:    false,
+			})
+		})
+
+		It("env init should succeed", func() {
+			Expect(testEnvInitErr).NotTo(HaveOccurred())
+		})
+	})
+
+	Context("when init a svc", func() {
+		var (
+			svcInitErr error
+		)
+		BeforeAll(func() {
+			_, svcInitErr = cli.SvcInit(&client.SvcInitRequest{
+				Name:       svcName,
+				SvcType:    "Load Balanced Web Service",
+				Dockerfile: "./hello/Dockerfile",
+				SvcPort:    "3000",
+			})
+		})
+
+		It("svc init should succeed", func() {
+			Expect(svcInitErr).NotTo(HaveOccurred())
+		})
+
+		It("svc init should create svc manifests", func() {
+			Expect("./copilot/hello/manifest.yml").Should(BeAnExistingFile())
+		})
+
+		It("svc ls should list the service", func() {
+			svcList, svcListError := cli.SvcList(appName)
+			Expect(svcListError).NotTo(HaveOccurred())
+			Expect(len(svcList.Services)).To(Equal(1))
+
+			svcsByName := map[string]client.SvcDescription{}
+			for _, svc := range svcList.Services {
+				svcsByName[svc.Name] = svc
+			}
+
+			for _, svc := range []string{svcName} {
+				Expect(svcsByName[svc].AppName).To(Equal(appName))
+				Expect(svcsByName[svc].Name).To(Equal(svc))
+			}
+		})
+	})
+
+	Context("build and push sidecar image to ECR repo", func() {
+		var uri string
+		tag := "vortexstreet"
+		It("create new ECR repo for sidecar", func() {
+			var b bytes.Buffer
+			err := command.Run("bash", []string{"-c", fmt.Sprintf("aws ecr create-repository --repository-name %s | jq -r .repository.repositoryUri", sidecarRepoName)}, command.Stdout(&b))
+			Expect(err).NotTo(HaveOccurred(), "create ECR repo for sidecar")
+			uri = strings.TrimSpace(b.String())
+			sidecarImageURI = fmt.Sprintf("%s:%s", uri, tag)
+		})
+		It("push sidecar image", func() {
+			var b bytes.Buffer
+			err := command.Run("aws", []string{"ecr", "get-login-password"}, command.Stdout(&b))
+			password := b.String()
+			Expect(err).NotTo(HaveOccurred(), "get ecr login password")
+			err = command.Run("docker", []string{"login", "-u", "AWS", "--password-stdin", uri}, command.Stdin(strings.NewReader(password)))
+			Expect(err).NotTo(HaveOccurred(), "docker login")
+			err = command.Run("docker", []string{"build", "-t", sidecarImageURI, "./nginx"})
+			Expect(err).NotTo(HaveOccurred(), "build sidecar image")
+			err = command.Run("docker", []string{"push", sidecarImageURI})
+			Expect(err).NotTo(HaveOccurred(), "push to ECR repo")
```

Comment:
Do you think this should be a BeforeAll of the 'write manifest file' section?

Reply:
Yeah that's a good idea!

Reply:
i tried to use BeforeAll but it seems like inside of beforeAll it doesn't allow me to have multiple It(). so i'll leave it as it is now.